osquery/events/darwin/endpointsecurity.h

```diff
@@ -117,6 +141,47 @@ class EndpointSecurityPublisher
   bool es_client_success_{false};
 };
 
+class EndpointSecurityFileEventPublisher
+    : public EventPublisher<EndpointSecurityFileSubscriptionContext,
+                            EndpointSecurityFileEventContext> {
+  DECLARE_PUBLISHER("endpointsecurity_fim");
+
+ public:
+  explicit EndpointSecurityFileEventPublisher(
+      const std::string& name = "EndpointSecurityFileEventPublisher")
+      : EventPublisher() {
+    runnable_name_ = name;
+  }
+
+  Status setUp() override API_AVAILABLE(macos(10.15));
+
+  void configure() override API_AVAILABLE(macos(10.15));
+
+  void tearDown() override API_AVAILABLE(macos(10.15));
+
+  Status run() override API_AVAILABLE(macos(10.15)) {
+    return Status::success();
+  }
+
+  bool shouldFire(const EndpointSecurityFileSubscriptionContextRef& sc,
+                  const EndpointSecurityFileEventContextRef& ec) const override
+      API_AVAILABLE(macos(10.15));
+
+  virtual ~EndpointSecurityFileEventPublisher() API_AVAILABLE(macos(10.15)) {
+    tearDown();
+  }
+
+ public:
+  static void handleMessage(const es_message_t* message)
+      API_AVAILABLE(macos(10.15));
+
+ private:
+  es_client_s* es_file_client_{nullptr};
```

Comment:
consider use std::unique_ptr with custom deleter that that calls to es_delete_client()

Reply:
I have had panics with `es_delete_client` on macOS 10.15 Catalina and this has been an issue that others have encountered too, we still want to support users on Catalina, doing it in a bit of a brute way. But open to suggestions/guarding it better, thoughts?